File: 55. Jump Game/canJump.py
class canJump:
    def solution(self, nums: List[int]) -> bool:
        maxJump = 0
        for i in range(len(nums)):
            if i <= maxJump:
                maxJump = max(maxJump, i + nums[i])
            if maxJump >= len(nums) - 1:
                return True
        return False

File: 55. Jump Game/canJump.java
class canJump {
    public boolean solution(int[] nums) {
        int maxPosition = 0;
        for (int i = 0; i < nums.length; i ++) {
            if (i <= maxPosition) {
                maxPosition = Math.max(maxPosition, i + nums[i]);
            }
            if (maxPosition >= nums.length - 1) {
                return true;
            }
        }
        return false;
    }
}